src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -117,21 +147,41 @@ public function get( $url ) {
 	 */
 	private function get_expiry_time( Response $response ) {
 		if ( $this->use_cache_control && $response->hasHeader( self::CACHE_CONTROL ) ) {
-			$max_age              = 0;
-			$cache_control_string = $response->getHeader( self::CACHE_CONTROL );
-			$cache_control_parts  = array_map( 'trim', explode( ',', $cache_control_string ) );
+			$max_age = $this->get_max_age( $response->getHeader( self::CACHE_CONTROL ) );
+
+			if ( $max_age >= $this->min_expiry ) {
+				return new DateTimeImmutable( "+ {$max_age} seconds" );
+			}
```

Comment:
Minor thing. Should this not rather be:

```suggestion
			$expiry = max( $this->min_expiry, $this->get_max_age( $response->getHeader( self::CACHE_CONTROL ) ) );
			return new DateTimeImmutable( "+ {$expiry} seconds" );
```

This will prevent a `max-age` of 10 minutes from automatically extending to the default expiry of 1 month, rather than the minimum expiry of 1 day.

Reply:
Ah, I see the problem. However, I don't think this is the solution, as you've eliminated the default expiry of 1 month altogether now. I'll update the PR to avoid the case you're trying to fix.

Reply:
Hmm, no, I' got confused with the alignment here in the PR. Let me take a look at my IDE instead...

Reply:
Is the default expiry needed then in this case? The default expiry is only needed when `max-age` is not used.

Reply:
Yes, indeed.

Reply:
However, the above code runs if `cache-control` is used. Is it guaranteed that it always contains a `max-age` directive?

Reply:
Right, it wouldn't. But then `get_max_age` would return `0` and then automatically the `min_expiry` would end up getting used.